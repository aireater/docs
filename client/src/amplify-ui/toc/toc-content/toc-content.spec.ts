import {AmplifyTOCContent} from "./toc-content";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-toc-content", () => {
  describe("Component logic", () => {
    let tocContent: AmplifyTOCContent;
    beforeEach(() => (tocContent = new AmplifyTOCContent()));

    it("should init `setContent` as `undefined`", () =>
      expect(tocContent.setContent).toBeUndefined());

    it("should init `content` as truthy", () =>
      expect(tocContent.content).toBeTruthy());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifyTOCContent],
            html: `<amplify-toc-content />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
