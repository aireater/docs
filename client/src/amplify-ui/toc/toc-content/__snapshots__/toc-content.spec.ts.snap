// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-toc-content Render logic should render 1`] = `MockHTMLElement {}`;
